src/app/Fake.Tools.SignTool/SignTool.fs

```diff
@@ -0,0 +1,403 @@
+        @"[ProgramFilesX86]\Windows Kits\10\bin\**\x86"
+    ]
+    ProcessUtils.tryFindFile winSdksDirs "signtool.exe"
+
+
+let private timestamping timestamp = seq {
+    match timestamp with
+    | Some t ->
+        let serverUrl = t.ServerUrl |> Option.defaultValue "http://timestamp.digicert.com"
+        yield match t.Algorithm with
+              | Some SHA1 -> sprintf "/t \"%s\"" serverUrl
+              | Some SHA256 | None -> sprintf "/tr \"%s\" /td \"sha256\"" serverUrl
+    | None ->
+        ()
+}
+let private digesting = function
+    | Some SHA1 -> "/fd \"sha1\""
+    | Some SHA256 | None -> "/fd \"sha256\""
+let private verbosing verbosity = seq {
+    match verbosity with
+    | Some v -> match v with
+                | Quiet -> yield "/q"
+                | Verbose -> yield "/v"
+    | None -> ()
+}
+let private yieldIfTrue yieldVal value = seq {
+    match value with
+    | Some _ -> yield sprintf "/%s" yieldVal
+    | None -> ()
+}
+let private yieldIfSome yieldGen value = seq {
+    match value with
+    | Some v -> yield yieldGen v
+    | None -> ()
+}
+let private sarg = sprintf "/%s \"%s\""
+let private iarg = sprintf "/%s %i"
+
+
+/// run the sign command using a runner
+let internal signInternal runner signtoolexeLocator (options: SignOptions) (files: seq<string>) =
+    let signtoolOptions = seq {
+        yield! yieldIfTrue "debug" options.Debug
+        yield! verbosing options.Verbosity
+        yield! yieldIfTrue "as" options.AppendSignature
+
+        match options.Certificate with
+        | File f ->
+            yield sarg "f" f.Path
+            yield! yieldIfSome (sarg "p") f.Password
+            yield! yieldIfSome (sarg "csp") f.CspName
+            yield! yieldIfSome (sarg "kc") f.PrivateKeyKey
+        | Store s ->
+            yield! yieldIfTrue "a" s.AutomaticallySelectCertificate
+            yield! yieldIfSome (sarg "i") s.IssuerName
+            yield! yieldIfSome (sarg "n") s.SubjectName
+            yield! yieldIfSome (sarg "r") s.RootSubjectName
+            yield! yieldIfSome (sarg "s") s.StoreName
+            yield! yieldIfSome (sarg "sha1") s.Hash
+            yield! yieldIfTrue "sm" s.UseComputerStore
+
+        yield digesting options.DigestAlgorithm
+        yield! timestamping options.TimeStamp
+        yield! yieldIfSome (sarg "ac") options.AdditionalCertificate
+        yield! yieldIfSome (sarg "c") options.CertificateTemplateName
+        yield! yieldIfSome (sarg "d") options.Description
+        yield! yieldIfSome (sarg "u") options.EnhancedKeyUsage
+        yield! yieldIfTrue "uw" options.EnhancedKeyUsageW
+    }
+    // hide password in trace output
+    match Context.isFakeContext (), options.Certificate with
+    | true, File f ->
+        if f.Password.IsSome then
+            // surround in quotes to lower chances of replacing non-password occurences of password-string
+            TraceSecrets.register "\"<PASSWORD>\"" (sprintf "\"%s\"" f.Password.Value)
```

Comment:
I guess you mean if a very simple password is used and replaced elsewhere?
This is a known problem, even on build servers which support secret variables (like azure devops)
I'm not sure we need special casing here, this might lead to the password being printed when it is not quoted?

Reply:
I ran into this when I was using a test certificate where the password was also in the file name (to make sure I didn't forget it) - part of the file name was replaced with the registered replacement which looked pretty weird.
The implementation makes sure it is always quoted and there is a test for it too, so erroneously displaying the password should not be a problem.